src/App.tsx

```
@@ -1,37 +1,48 @@
 import React from 'react';
 import './App.scss';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+import { Clock } from './Clock';
 
-  return `Clock-${value}`;
+function getRandomName(): string {
+  return `Clock-${Date.now().toString().slice(-4)}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">{clockName}</strong>
-
-        {' time is '}
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component<{}, State> {
+  private timerId?: ReturnType<typeof setInterval>;
+
+  state: State = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  componentDidMount() {
```

This violates requirement 1.10: 'On document right-click (contextmenu event), prevent default context menu and hide the Clock (set hasClock to false).' and checklist item #2.

You are not preventing the default context menu behavior or hiding the Clock on right-click/contextmenu.